scripts/thumbnail-generator/generate_thumbnail.py

```diff
@@ -0,0 +1,302 @@
+
+def generate_thumbnail(usd_file, verbose, extension):
+    if verbose: 
+        print("Step 1: Setting up the camera...")
+    
+    subject_stage = Usd.Stage.Open(usd_file)
+    subject_file = usd_file
+
+    if (UsdGeom.GetStageUpAxis(subject_stage) == 'Z'):
+        subject_stage = generate_y_up_stage(subject_stage, usd_file)
+        subject_file = 'y_up.usda'
+
+    setup_camera(subject_stage, subject_file)
+    
+    if verbose:
+        print("Step 2: Taking the snapshot...")
+    
+    Path("renders").mkdir(parents=True, exist_ok=True)
+
+    image_path = os.path.join("renders", create_image_filename(usd_file, extension)).replace("\\", "/")
+    image_name = take_snapshot(image_path)
+
+    return image_name
+
+def generate_y_up_stage(stage, usd_file):
+    y_up_stage = Usd.Stage.CreateNew('y_up.usda')
+    new_top_level = UsdGeom.Xform.Define(y_up_stage, '/Root')
+
+    for prim in stage.GetPseudoRoot().GetChildren():
+        if prim != new_top_level.GetPrim():
+            new_prim = y_up_stage.DefinePrim(new_top_level.GetPath().AppendChild(prim.GetName()), prim.GetTypeName())
+            new_prim.SetActive(True)
+            new_prim.GetReferences().AddReference(usd_file, prim.GetPath())
+
+    # Iterate over all prims in the stage
+    for prim in Usd.PrimRange(stage.GetPseudoRoot()):
+        # Check if the prim has the MaterialBindingAPI applied
+        if prim.HasAPI(UsdShade.MaterialBindingAPI):
+            # Create a MaterialBindingAPI for the mesh prim
+            binding = UsdShade.MaterialBindingAPI(prim)
+            bound_material, binding_rel = binding.ComputeBoundMaterial()
+           
+            # get path to new prim (prepend top layer)
+            root_mesh_prim = y_up_stage.GetPrimAtPath('/Root' + str(prim.GetPath()))
+            
+            root_material_prim = y_up_stage.GetPrimAtPath('/Root' + str(bound_material.GetPath()))
+            root_material = UsdShade.Material(root_material_prim)
+
+            materialBindingAPI = UsdShade.MaterialBindingAPI(root_mesh_prim)
+            materialBindingAPI.Bind(root_material)
+
+    # Apply the rotation to the parent prim
+    UsdGeom.Xformable(new_top_level).AddRotateXOp().Set(270)
+
+    y_up_stage.Save()
+    return y_up_stage
+
+
+def setup_camera(subject_stage, usd_file):
+    camera_stage = create_camera()
+    move_camera(camera_stage, subject_stage)
+
+    # check if string is not empty
+    if args.dome_light:
+        add_domelight(camera_stage)
+
+    sublayer_subject(camera_stage, usd_file)
+
+def create_camera():
+    stage = Usd.Stage.CreateNew('camera.usda')
+
+    # Set metadata on the stage.
+    stage.SetDefaultPrim(stage.DefinePrim('/ThumbnailGenerator', 'Xform'))
+    stage.SetMetadata('metersPerUnit', 0.01)
+
+    # Define the "MainCamera" under the "ThumbnailGenerator".
+    camera = UsdGeom.Camera.Define(stage, '/ThumbnailGenerator/MainCamera')
+
+    # Set the camera attributes.
+    camera.CreateFocalLengthAttr(50)
```

Comment:
Any of these attributes that match the defaults for a camera prim don't need to be here. Also, setting the focus distance should (if anything) be using the distance to the center of the bbox of the asset probably? But I'm not sure there's any point in setting it since the fstop is set to zero which should prevent any depth of field effect based on the focus distance...

Reply:
Good catch, will update